src/test/java/hudson/scm/IntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+                    os.write(b);
+                    os.flush();
+                }
+            }
+            @Override public void run() {
+                try {
+                    while (true) {
+                        final Socket s;
+                        try {
+                            s = sock.accept();
+                        } catch (SocketException x) {
+                            // Socket closed?
+                            break;
+                        }
+                        LOGGER.info("accepted client connection");
+                        ProcessBuilder pb = new ProcessBuilder("cvs", "-f", "--allow-root=" + repo, "pserver");
+                        final Process server = pb.start();
+                        new Thread("printing errors") {
+                            @Override public void run() {
+                                try {
+                                    copy(server.getErrorStream(), System.err);
+                                } catch (IOException x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        }.start();
+                        new Thread("sending output") {
+                            @Override public void run() {
+                                try {
+                                    copy(server.getInputStream(), /*new TeeOutputStream(*/s.getOutputStream()/*, System.err)*/);
+                                } catch (SocketException x) {
+                                    // Broken pipe? Ignore.
+                                } catch (IOException x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        }.start();
+                        new Thread("accepting input") {
+                            @Override public void run() {
+                                try {
+                                    InputStream is = s.getInputStream();
+                                    OutputStream os = server.getOutputStream();
+                                    copy(is, /*new TeeOutputStream(*/os/*, System.err)*/);
+                                } catch (IOException x) {
+                                    x.printStackTrace();
+                                }
+                            }
+                        }.start();
+                        new Thread("wait for exit") {
+                            @Override public void run() {
+                                try {
+                                    int r = server.waitFor();
+                                    if (r != 0) {
+                                        LOGGER.log(Level.INFO, "server exited with status {0}", r);
+                                    }
+                                } catch (Exception x) {
+                                    x.printStackTrace();
```

Comment:
Log the exception rather than printStackTrace - and any reason you're catching such a broad exception?